src/main/java/nell/Nell.java

```diff
@@ -0,0 +1,346 @@
+    }
+
+    private static void loadFromFile() {
+        try {
+            File directory = new File("data");
+            File dataFile = new File("data/data.txt");
+
+            if (!dataFile.exists()) {
+                directory.mkdir();
+                dataFile.createNewFile();
+            }
+
+            Scanner fileScanner = new Scanner(dataFile);
+            while (fileScanner.hasNext()) {
+                String fileTask = fileScanner.nextLine();
+                loadTask(fileTask);
+            }
+        } catch (IOException e) {
+            System.out.println("Data not loaded - File not found");
+        }
+    }
+
+    /**
+     * Parses a task from the file
+     *
+     * @param taskLine The file line containing the task
+     */
+    private static void loadTask(String taskLine) {
+        String[] taskParameters = taskLine.split("\\|");
+        String taskType = taskParameters[0];
+        String taskDescription = taskParameters[2];
+
+        boolean taskIsDone = false;
+        if (taskParameters[1].equals("X"))
+        {
+            taskIsDone = true;
+        }
+
+        switch (taskType) {
+        case "T":
+            ToDo toDoToAdd = new ToDo(taskDescription, taskIsDone);
+            tasks.loadTask(toDoToAdd);
+            break;
+
+        case "D":
+            String deadlineBy = taskParameters[3];
+            Deadline deadlineToAdd = new Deadline(taskDescription, taskIsDone, deadlineBy);
+            tasks.loadTask(deadlineToAdd);
+            break;
+
+        case "E":
+            String eventFrom = taskParameters[3];
+            String eventTo = taskParameters[4];
+            Event eventToAdd = new Event(taskDescription, taskIsDone, eventFrom, eventTo);
+            tasks.loadTask(eventToAdd);
+            break;
+
+        default:
+            break;
+        }
+    }
+
+    /**
+     * Saves the task list data in a file
+     */
+    private static void saveToFile() {
+        try {
+            String filePath = "data/data.txt";
+            tasks.writeListToFile(filePath);
+        } catch (IOException e) {
+            System.out.println("   Data not saved due to error");
+        }
+    }
+
+    /**
+     * Greet the user upon program startup
+     */
+    private static void greetUser() {
+        System.out.println("Hello! I'm Nell!");
+        System.out.println("What can I do for you?");
+    }
+
+    /**
+     * Get commands from user and execute commands received
+     */
+    private static void getCommands() {
+        // Initialises scanner to take in user input
+        Scanner input = new Scanner(System.in);
+
+        boolean isGettingCommands = true;
+
+        while (isGettingCommands) {
+            // Get user command and respond accordingly
+            String command = input.nextLine();
+            String[] commandWords = command.split(" ", 2);
+            switch (commandWords[0]) {
+            case "bye":
+                sayBye();
+                saveToFile();
```

Comment:
Try to save tasks at the end of each operation, so that even when the app is not closed properly it retains the changes made.